openassessment/fileupload/backends/gcs.py

```diff
@@ -0,0 +1,63 @@
+"""GCS Bucket File Upload Backend."""
+
+from google.cloud import storage
+import logging
+
+from ..exceptions import FileUploadInternalError
+from .base import BaseBackend
+
+log = logging.getLogger("openassessment.fileupload.api")  # pylint: disable=invalid-name
+
+
+class GCSBackend(BaseBackend):
+    """ GCS Bucket File Upload Backend. """
+
+    def get_upload_url(self, key, content_type):
+        """Get a signed URL for uploading a file to GCS"""
+        bucket_name, key_name = self._retrieve_parameters(key)
+        try:
+            blob = get_blob_object(bucket_name, key_name)
+            url = blob.generate_signed_url(
+                version="v4",
+                expiration=self.UPLOAD_URL_TIMEOUT,
+                method="PUT",
+                content_type=content_type,
+            )
+            return url
+        except Exception as ex:
+            log.exception("An internal exception occurred while generating an upload URL.")
+            raise FileUploadInternalError(ex) from ex
+
+    def get_download_url(self, key):
+        """Get a signed URL for downloading a file from GCS"""
+        bucket_name, key_name = self._retrieve_parameters(key)
+        try:
+            blob = get_blob_object(bucket_name, key_name)
+            if not blob.exists():
+                return ""
+            url = blob.generate_signed_url(
+                version="v4",
+                expiration=self.DOWNLOAD_URL_TIMEOUT,
+                method="GET",
+            )
+            return url
+        except Exception as ex:
+            log.exception("An internal exception occurred while generating a download URL.")
+            raise FileUploadInternalError(ex) from ex
+
+    def remove_file(self, key):
+        """Remove a file from GCS"""
+        bucket_name, key_name = self._retrieve_parameters(key)
+        blob = get_blob_object(bucket_name, key_name)
+        if blob.exists():
+            blob.delete()
+            return True
+        return False
+
+
+def get_blob_object(bucket_name, key_name):
+    """Get a blob object from GCS"""
+    client = storage.Client()
+    bucket = client.bucket(bucket_name)
+    blob = bucket.blob(key_name)
```

Comment:
We don't need to temporarily store the result in a local `blob` variable. Rather we can, 

```
return bucket.blob(key_name)
```